tests/testvarint.nim

```diff
@@ -147,6 +292,23 @@ suite "Variable integer test suite":
         check:
           PB.getUVarint(buffer, length, value) == VarintStatus.Overflow
 
+  test "[ProtoBuf] Test vectors":
+    # The test vectors which was obtained at:
+    # https://github.com/dermesser/integer-encoding-rs/blob/master/src/varint_tests.rs
+    # https://github.com/That3Percent/zigzag/blob/master/src/lib.rs
+    check:
+      PB.encodeVarint(0'u64) == @[0x00'u8]
+      PB.encodeVarint(0'u32) == @[0x00'u8]
+      PB.encodeVarint(0'i64) == @[0x00'u8]
+      PB.encodeVarint(0'i32) == @[0x00'u8]
```

Comment:
when we discussed, this is the trap I was worried about: that "normal" integers are passed in without cast, while zigzag requires cast - would it make sense that there's a distinct type for cast-style encoding as well? otherwise, it will happen that callers forget and mix the styles

Reply:
So lets change behavior and allow people to use `int32` and `int64` for zigzag encoding and `sint32` and `sint64` for cast encoding.

Reply:
Or you want equal behavior for all types of encoding?

Reply:
equal for all is appropriate I think - not sure if there's one of `cast` and `zigzag` that is more "standard" in the protobuf community

Reply:
in part, this is why my first thought went to naming - because there isn't a clear "natural" choice

Reply:
@arnetheduck when i have searching for test vectors, i have found 0 test vectors for `cast` and little amount for `zigzag`. So i think the most common is `zigzag`.

Reply:
So i'm proposing
```nim
type
  ## Integer types for "zigzag" encoding
  zint32 = distinct int32
  zint64 = distinct int64
  zint = distinct int
  ## Integer types for "cast" encoding
  hint32 = distinct int32
  hint64 = distinct int64
  hint = distinct int
```

Reply:
sure, that should be safe.